src/app/pages/task-calendar/snapshot/snapshot-form/snapshot-form.component.ts

```diff
@@ -1,169 +1,172 @@
 import { Component } from '@angular/core';
-import { Router } from '@angular/router';
-
+import { Router,ActivatedRoute} from '@angular/router';
+import { Validators } from '@angular/forms';
 import * as _ from 'lodash';
 
-import { EntityFormComponent } from '../../../common/entity/entity-form';
 import { FieldConfig } from '../../../common/entity/entity-form/models/field-config.interface';
-import { TaskService, UserService } from '../../../../services/';
-import { EntityFormService } from '../../../common/entity/entity-form/services/entity-form.service';
+import { TaskService } from '../../../../services/';
 import { EntityUtils } from '../../../common/entity/utils';
 import helptext from '../../../../helptext/task-calendar/snapshot/snapshot-form';
 
-
 @Component({
   selector: 'cron-snapshot-task-add',
   template: `<entity-form [conf]="this"></entity-form>`,
-  providers: [TaskService, UserService, EntityFormService]
+  providers: [TaskService]
 })
 export class SnapshotFormComponent {
 
-  protected resource_name: string = 'storage/task';
+  protected queryCall = "pool.snapshottask.query";
+  protected addCall = "pool.snapshottask.create";
+  protected editCall = "pool.snapshottask.update";
+  protected customFilter: Array<any> = [[["id", "="]]];
   protected route_success: string[] = ['tasks', 'snapshot'];
-  protected entityForm: EntityFormComponent;
   protected isEntity: boolean = true;
+  protected pk: any;
 
   public fieldConfig: FieldConfig[] = [{
     type: 'select',
-    name: 'task_filesystem',
-    placeholder: helptext.task_filesystem_placeholder,
-    tooltip: helptext.task_filesystem_tooltip,
+    name: 'dataset',
+    placeholder: helptext.dataset_placeholder,
+    tooltip: helptext.dataset_tooltip,
     options: [],
     required: true,
-    validation : helptext.task_filesystem_validation
+    validation: [Validators.required],
   }, {
     type: 'checkbox',
-    name: 'task_recursive',
-    placeholder: helptext.task_recursive_placeholder,
-    tooltip: helptext.task_recursive_tooltip
+    name: 'recursive',
+    placeholder: helptext.recursive_placeholder,
+    tooltip: helptext.recursive_tooltip,
+    value: false,
   }, {
-    placeholder: helptext.task_ret_count_placeholder,
+    type: 'textarea',
+    name: 'exclude',
+    placeholder: helptext.exclude_placeholder,
+    tooltip: helptext.exclude_tooltip
+  }, {
+    placeholder: helptext.lifetime_value_placeholder,
     type: 'input',
-    name: 'task_ret_count',
+    name: 'lifetime_value',
     inputType: 'number',
     class: 'inline',
     value: 2,
-    validation: helptext.task_ret_count_validation
+    validation: [Validators.min(0)]
   }, {
     type: 'select',
-    name: 'task_ret_unit',
-    tooltip: helptext.task_ret_unit_tooltip,
+    name: 'lifetime_unit',
+    tooltip: helptext.lifetime_unit_tooltip,
     options: [{
-      label: 'Hours',
-      value: 'hour',
+      label: 'Hour(s)',
```

Comment:
Please don't use (s) to pluralize unless this is a legal form.
```suggestion
      label: 'Hours',
```